server/bin/basket-proxy-server.js

```diff
@@ -0,0 +1,180 @@
+
+    if (! authHeader) {
+      logger.error('auth.missing-authorization-header');
+      res.status(400).json(errorResponse('missing authorization header', BASKET_ERRORS.USAGE_ERROR));
+      return;
+    }
+
+    var token = authHeader.replace(/^Bearer /, '');
+
+    request.post({
+      url: VERIFY_URL,
+      json: {
+        token: token
+      }
+    }, function (err, _, body) {
+      // TODO use an actual error format
+      if (err) {
+        logger.error('auth.error', err);
+        res.status(400).json(errorResponse(err, BASKET_ERRORS.UNKNOWN_ERROR));
+        return;
+      }
+
+      if (body.code >= 400) {
+        logger.error('auth.unauthorized', body);
+        res.status(body.code).json(errorResponse('unauthorized', BASKET_ERRORS.AUTH_ERROR));
+        return;
+      }
+
+      logger.debug('auth.valid', body);
+
+      res.locals.creds = body;
+
+      next();
+    });
+  };
+}
+
+// Send a request to the Basket backend
+function basketRequest(path, method, params, done) {
+  var req = request({
+    url: API_URL + path,
+    strictSSL: true,
+    method: method,
+    headers: {
+      'X-API-Key': API_KEY
+    },
+    form: params
+  }, done);
+
+  return req;
+}
+
+function initApp() {
+  var app = express();
+  app.use(bodyParser.json());
+
+  app.use(cors({
+    origin: CORS_ORIGIN
+  }));
+
+  app.use(verifyOAuthToken());
+
+  app.get('/lookup-user', function (req, res) {
+    var params = req.body;
+    var creds = res.locals.creds;
+
+    basketRequest('/lookup-user/?email=' + creds.email, 'get', params).pipe(res);
+  });
+
+  app.post('/subscribe', function (req, res) {
+    var params = req.body;
+    params.email = res.locals.creds.email;
+
+    basketRequest('/subscribe/', 'post', params).pipe(res);
+  });
+
+  app.post('/unsubscribe', function (req, res) {
+    var creds = res.locals.creds;
+    basketRequest('/lookup-user/?email=' + creds.email, 'get', {}, function (lookupError, httpRequest, body) {
+      if (lookupError) {
+        logger.error('lookup-user.error', lookupError);
+        res.status(400).json(errorResponse(lookupError, BASKET_ERRORS.UNKNOWN_ERROR));
+        return;
+      }
+
+      var responseData;
+      try {
+        responseData = JSON.parse(body);
+      } catch (parseError) {
+        logger.error('lookup-user.cannot-parse-response', parseError);
+        res.status(400).json(errorResponse(parseError, BASKET_ERRORS.UNKNOWN_ERROR));
+        return;
+      }
+      if (responseData.status !== 'ok') {
+        logger.error('lookup-user status not ok: ' + responseData.status);
+        res.status(httpRequest.statusCode).json(responseData);
+        return;
+      }
+
+      var params = req.body;
+      params.email = creds.email;
+
+      basketRequest('/unsubscribe/' + responseData.token + '/', 'post', params).pipe(res);
+    });
+  });
+
+  return app;
+}
+
+function listen(app) {
+  var serverUrl = url.parse(config.get('marketing_email.api_url'));
```

Comment:
Per discussion with @jrgm in IRC; the `marketing_email.api_url` is the publicly-visible URL of this API.  For hosting purposes we will want to listen on a localhost port.  I suggest making a new `basket.proxy_url` config option to set the local listen address for this proxy to bind to, and keeping `marketing_email.api_url` as the public-facing one.

In practice, we'll host with some sort of nginx proxy so the settings might be something like:

  * `marketing_email.api_url: "https://accounts.firefox.com/basket/news"`
  * `basket.proxy_url: "http://localhost:3034"`

@zaach thoughts?